app/assets/javascripts/collection_editor.js

```diff
@@ -9,112 +9,119 @@
+
+        // Iterate through the pool of strings and for any string that
+        // contains the substring `q`, add it to the `matches` array
+        $.each(strs, function(i, str) {
+          if (substrRegex.test(str)) {
+            matches.push(str);
+          }
+        });
+        cb(matches);
+      };
+    },
+
+    autoComplete: function(){
+      this.$('.typeahead').typeahead({
+        hint: true,
+        highlight: true,
+        minLength: 1
+      }, {
+        name: 'fields',
+        source: this.substringMatcher(this.whitelist)
+      });
+    },
 
-      cb(matches);
-    };
-  };
-
-  var autoComplete = function(){
-    $('.typeahead').typeahead({
-      hint: true,
-      highlight: true,
-      minLength: 1
-    }, {
-      name: 'fields',
-      source: substringMatcher(whitelist)
-    });
-  };
-
-  var makeSortable = function(){
-    $( ".list-group.sortable" ).sortable();
-  };
-
-  var updateSort = function( event, ui, a, b ) {
-    var ids = ui.item.parent().
-      children().
-        map(function(i, el){
-          return $(el).data('id');
-        }).get().join();
-
-    $('#form_element_ids').val(ids);
-    $('form#sort-collection-elements').submit();
-  };
-
-  var bindHandlers = function(){
-    $('.collection-editor #new_collection_element').on('ajax:success', function(e, resp, c){
-      $('.list-group').append(resp);
-      makeSortable();
-    });
+    makeSortable: function(){
+      this.$( ".list-group.sortable" ).sortable();
+    },
 
-    $('.collection-editor').on('ajax:success', "a[data-method=delete]", function(){
-      $(this).parents('.list-group-item').fadeOut();
-    });
+    updateSort: function( event, ui, a, b ) {
+      var ids = ui.item.parent().
+        children().
+          map(function(i, el){
+            return $(el).data('id');
+          }).get().join();
+
+      this.$('#form_element_ids').val(ids);
+      this.$('form#sort-collection-elements').submit();
+    },
 
-    $( ".collection-editor" ).on( "sortupdate", updateSort );
+    newElementAdded: function(e, resp, c) {
+      this.$('.list-group').append(resp);
+      this.makeSortable();
+    },
 
-    $('#change-form-template').on('ajax:success', function(e, resp) {
-      $('.forms-edit').html(resp.html);
-      makeSortable();
+    templateChanged: function(e, resp) {
+      this.$('.forms-edit').html(resp.html);
+      this.makeSortable();
 
       // Updates the inline form's action URL with the new form ID.
-      $('#sort-collection-elements, #new_collection_element').each(function(i, el){
+      this.$('#sort-collection-elements, #new_collection_element').each(function(i, el){
         var action = $(el).attr('action').replace(/\d+/, resp.form_id);
         $(el).attr('action', action);
       });
+    }
+  });
+
+  const configureCollections = function() {
+    $('.collection-editor').each(function(ii, el){
+      let $el = $(el);
+      if( $el.data('js-inited') != true) {
```

Comment:
you can do that but I'm not sure being less explicit is better... data attributes are often strings so I want to be very clear what I mean.
```js
>>> "true" != true
true
>>> !"true"
false
```